adapter/bkm_express.go

```diff
@@ -41,7 +41,21 @@ func (api *BkmExpress) Complete(ctx context.Context, request CompleteBkmExpressR
 }
 
 func (api *BkmExpress) RetrievePayment(ctx context.Context, ticketId string) (*PaymentResponse, error) {
-	newRequest, err := api.Client.NewRequest(ctx, http.MethodGet, fmt.Sprintf("/payment/v1/card-payments/%s", ticketId), nil)
+	newRequest, err := api.Client.NewRequest(ctx, http.MethodGet, fmt.Sprintf("/payment/v1/bkm-express/payments/%s", ticketId), nil)
+	if err != nil {
+		return nil, err
```

Comment:
In order to distinguish errors in this function, it could be best if you add error context. Instead of returning err only, you could use 

`return nil, fmt.Errorf("failed to create request. %v", err)` for the first one, and

`return nil, fmt.Errorf("failed to retrieve payment. %v", err)`

By doing this, you will have proper stack trace in the caller functions.